src/parser/get_map.c

```diff
@@ -11,30 +11,73 @@
 /* ************************************************************************** */
 
 #include "CBDerror.h"
-#include "CBDparser.h"
 #include "Cub3d.h"
+#include "CBDparser.h"
 
 #include "libft.h"
+
 #include <sys/types.h>
 
-bool	get_map(t_cub3d *info, char **line, ssize_t *idx)
+static size_t	get_line_amount(char *line, ssize_t *idx)
 {
-	ssize_t	i;
+	size_t	line_count;
+	size_t	i;
 
+	line_count = 0;
 	i = *idx;
 	while (line[i])
+	{
+		if (line[i] == '\n')
+			line_count++;
 		i++;
-	info->map = ft_calloc((i - *idx) + 1, sizeof(char *));
-	if (!info->map)
-		return (cbd_error(ERR_MEMORY), FAILURE);
+	}
+	return (line_count);
+}
+
+static char	*extract_line(char *line, ssize_t *idx)
+{
+	char	*ret;
+	size_t	len;
+
+	len = *idx;
+	while (line[len] && line[len++] != '\n')
+		;
+	len -= *idx;
+	ret = ft_substr(line, *idx, len);
+	*idx += len;
```

Comment:
```suggestion
	ret = ft_substr(line, *idx, len - *idx);
	*idx = len;
```

Reply:
This looks good.
There is a while loop in this same function that is a bit cursed:
(
while (line[len] && line[len++] != '\n')
    ;
)